src/picterra/client.py

```diff
@@ -82,22 +83,38 @@ def _wait_until_operation_completes(self, operation_response):
                 raise APIError('Operation %s failed' % operation_id)
             time.sleep(poll_interval)
 
-    def _paginate_through_list(self, resource_endpoint: str):
+    def _paginate_through_list(self, resource_endpoint: str, params: dict = {}):
+        # We explicit page number, even if it is not needed
         url = self._api_url('%s/?page_number=1' % resource_endpoint)
+        # Store list data, page (for debugging, future limit), total count (same as latter)
         data = []
+        page = 0
+        count = 0
+        # Loop over all the pages
         while url:
-            logger.debug('Paginating through %s list at page %s' % (resource_endpoint, url))
-            resp = self.sess.get(url)
+            page += 1
+            logger.debug('Paginating through %s list at page %s' % (resource_endpoint, page))
+            if page == 1:
+                # This special case is necessary because at the first request we must set the
+                # non-pagination querystring, while afterwards the "next" already has it
+                resp = self.sess.get(url, params=params)
+            else:
+                # From "next"
+                resp = self.sess.get(url)
+            # Error management
             if not resp.ok:
                 raise APIError(resp.text)
+            # Save page data and get reference to next (if any)
             r = resp.json()
             url = r['next']
             count = r['count']
             data += r['results']
-        assert len(data) == count
+        assert len(data) == count, 'Error in pagination [%d!=%d]' % (len(data), count)
         return data
 
-    def upload_raster(self, filename, name, folder_id=None, captured_at=None):
+    def upload_raster(
+        self, filename: str, name: str, folder_id: str = '', captured_at: str = ''
```

Comment:
Use `None` instead of empty strings for unset parameters.

Reply:
Checked